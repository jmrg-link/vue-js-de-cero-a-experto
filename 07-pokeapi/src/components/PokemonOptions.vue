<template >
    <div class="options-container">
        <ul>
            <li v-for="pokemon in pokemons" :key="pokemon.id" @click="$emit('selection',pokemon.id)">{{ pokemon.name }}</li>
        </ul>
    </div>
</template>

<script>
export default {
  props:{
    pokemons:{
      type:Array,
      required:true
    }
  }
}
</script>

<style scoped>
/* Pokemon Options */
ul {
    list-style-type: none;
}
li {
    background-color: white;
    border-radius: 6px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    padding:5px;
    margin-bottom: 10px;
    width: 375px;
    font-size:18px;
}

li:hover {
    background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
    display: flex;
    justify-content: center;
}
</style>