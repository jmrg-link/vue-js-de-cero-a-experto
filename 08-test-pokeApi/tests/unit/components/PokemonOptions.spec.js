import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import { pokemons } from "../mocks/pokemons.mock";

describe("PokemonOptions component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons,
      },
    });
  });

  test("Debe hacer match con el snapshot", () => {
    //console.log(wrapper.html());
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de mostrar las 4 opciones correctamente ', () => {
    const liTags = wrapper.findAll('li')
    expect(liTags.length).toBe(4)

    expect(liTags[0].text()).toBe('Pikachu')
    expect(liTags[1].text()).toBe('charmander')
    expect(liTags[2].text()).toBe('venusaur')
    expect(liTags[3].text()).toBe('mew')

  })

  test('debe de emitir "slection" con sus respectivos parametros al hacer click', () => {
    const [ li1,li2,li3,li4 ] = wrapper.findAll('li')
    li1.trigger('click')
    li2.trigger('click')
    li3.trigger('click')
    li4.trigger('click')
    console.log(wrapper.emitted('selection'))
    expect(wrapper.emitted('selection').length).toBe(4)
    expect(wrapper.emitted('selection')[0]).toStrictEqual([5])
    expect(wrapper.emitted('selection')[1]).toStrictEqual([10])
    expect(wrapper.emitted('selection')[2]).toStrictEqual([15])
    expect(wrapper.emitted('selection')[3]).toStrictEqual([20])
  })
  
  
});
