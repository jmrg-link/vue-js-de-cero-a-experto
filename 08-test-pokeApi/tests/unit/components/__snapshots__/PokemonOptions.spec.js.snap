// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component Debe hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>Pikachu</li>
    <li>charmander</li>
    <li>venusaur</li>
    <li>mew</li>
  </ul>
</div>
`;
